tests/test_check_unused_python_file.py

```diff
@@ -0,0 +1,60 @@
+import os
+from pathlib import Path
+from typing import List
+
+from oca_pre_commit_hooks.check_unused_python_file import main
+
+FILE_NAMES = ["res_partner", "project_task", "helpdesk_ticket"]
+
+
+def _gen_filepaths(filenames: List[str], basedir) -> List[str]:
+    return [f"{os.path.join(basedir, filename)}.py" for filename in filenames]
+
+
+def test_all_used_files(tmpdir):
+    filepaths = _gen_filepaths(FILE_NAMES, tmpdir)
+    init_file = os.path.join(tmpdir, "__init__.py")
+    with open(init_file, "w", encoding="utf-8") as init_fd:
+        init_fd.writelines([f"from . import {filename}\n" for filename in FILE_NAMES])
+
+    for filepath in filepaths:
+        Path(filepath).touch()
+
+    assert main(filepaths) == 0
+
+    Path(init_file).unlink()
+    with open(init_file, "w", encoding="utf-8") as init_fd:
+        init_fd.write(f"from . import {','.join(FILE_NAMES)}")
+
+    assert main(filepaths) == 0
+
+
+def test_all_unused_files(tmpdir):
+    filepaths = _gen_filepaths(FILE_NAMES, tmpdir)
+    init_file = os.path.join(tmpdir, "__init__.py")
+
+    Path(init_file).touch()
+    for filepath in filepaths:
+        Path(filepath).touch()
+
+    assert main(filepaths) == -1
+
+
+def test_complex_init(tmpdir):
```

Comment:
Could you add a unittest to silent valid cases?

I mean, imagine you have a odoo module to process a python file with a `dict` instead of a json (similar way  `__manifest__.py`) so you will have python files without importing but not default one

Reply:
> I think the best place to create python lints is pylint
> 
> It could be done using `def open` or using `def visit_module` in order to get all imported nodes for a particular module directory
> 
> Could we discuss it, please?

Yes, ideally Python code should be in pylint, I agree. I went with hooks since this PR was a joy to code haha, I feel the resulting code is clean and elegant, and the `files` and `exclude` options are helpful. I will look into moving it with the help of `visit_module`, I did not think of that before and it seems promising.